core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -240,6 +242,12 @@ static TableMetadata fromJson(TableOperations ops, InputFile file, JsonNode node
       specs = ImmutableList.of(PartitionSpecParser.fromJsonFields(
           schema, TableMetadata.INITIAL_SPEC_ID, node.get(PARTITION_SPEC)));
     }
+    // get the last spec
+    List<PartitionField> fields = specs.get(specs.size() - 1).fields();
+    if (fields.size() > 0) {
+      // get the last lastPartitionFieldId
+      lastAssignedPartitionFieldId = fields.get(fields.size() - 1).fieldId();
```

Comment:
Instead of getting the last ID, I think this should just keep track of the last assigned partition field ID, like the `last-column-id`. How about storing it as `last-partition-id`?